// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message NetworksecurityAlphaClientTlsPolicy {
  string name = 1;
  string description = 2;
  string create_time = 3;
  string update_time = 4;
  map<string, string> labels = 5;
  string sni = 6;
  NetworksecurityAlphaClientTlsPolicyClientCertificate client_certificate = 7;
  repeated NetworksecurityAlphaClientTlsPolicyServerValidationCa server_validation_ca = 8;
  string project = 9;
  string location = 10;
}

message NetworksecurityAlphaClientTlsPolicyClientCertificate {
  NetworksecurityAlphaClientTlsPolicyClientCertificateLocalFilepath local_filepath = 1;
  NetworksecurityAlphaClientTlsPolicyClientCertificateGrpcEndpoint grpc_endpoint = 2;
  NetworksecurityAlphaClientTlsPolicyClientCertificateCertificateProviderInstance certificate_provider_instance = 3;
}

message NetworksecurityAlphaClientTlsPolicyClientCertificateLocalFilepath {
  string certificate_path = 1;
  string private_key_path = 2;
}

message NetworksecurityAlphaClientTlsPolicyClientCertificateGrpcEndpoint {
  string target_uri = 1;
}

message NetworksecurityAlphaClientTlsPolicyClientCertificateCertificateProviderInstance {
  string plugin_instance = 1;
}

message NetworksecurityAlphaClientTlsPolicyServerValidationCa {
  string ca_cert_path = 1;
  NetworksecurityAlphaClientTlsPolicyServerValidationCaGrpcEndpoint grpc_endpoint = 2;
  NetworksecurityAlphaClientTlsPolicyServerValidationCaCertificateProviderInstance certificate_provider_instance = 3;
}

message NetworksecurityAlphaClientTlsPolicyServerValidationCaGrpcEndpoint {
  string target_uri = 1;
}

message NetworksecurityAlphaClientTlsPolicyServerValidationCaCertificateProviderInstance {
  string plugin_instance = 1;
}

message ApplyNetworksecurityAlphaClientTlsPolicyRequest {
  NetworksecurityAlphaClientTlsPolicy resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteNetworksecurityAlphaClientTlsPolicyRequest {
  string service_account_file = 1;
  NetworksecurityAlphaClientTlsPolicy resource = 2;
}

message ListNetworksecurityAlphaClientTlsPolicyRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListNetworksecurityAlphaClientTlsPolicyResponse {
  repeated NetworksecurityAlphaClientTlsPolicy items = 1;
}

service NetworksecurityAlphaClientTlsPolicyService {
  rpc ApplyNetworksecurityAlphaClientTlsPolicy(ApplyNetworksecurityAlphaClientTlsPolicyRequest) returns (NetworksecurityAlphaClientTlsPolicy);
  rpc DeleteNetworksecurityAlphaClientTlsPolicy(DeleteNetworksecurityAlphaClientTlsPolicyRequest) returns (google.protobuf.Empty);
  rpc ListNetworksecurityAlphaClientTlsPolicy(ListNetworksecurityAlphaClientTlsPolicyRequest) returns (ListNetworksecurityAlphaClientTlsPolicyResponse);
}
